receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -45,35 +76,66 @@ func (p *StatsDParser) Parse(line string) (*metricspb.Metric, error) {
 		return nil, fmt.Errorf("invalid <name>:<value> format: %s", parts[0])
 	}
 
-	metricName := parts[0][0:separatorIndex]
-	if metricName == "" {
+	result.name = parts[0][0:separatorIndex]
+	if result.name == "" {
 		return nil, errEmptyMetricName
 	}
-	metricValueString := parts[0][separatorIndex+1:]
-	if metricValueString == "" {
+	result.value = parts[0][separatorIndex+1:]
+	if result.value == "" {
 		return nil, errEmptyMetricValue
 	}
 
-	metricType := parts[1]
+	result.statsdMetricType = parts[1]
```

Comment:
Would it be a good idea to validate the metric type here? If it's invalid, we could return immediately rather than going through additional parts and then return an error while building the datapoint.